feat: Add Cloud SQL for PostgreSQL plugin with indexer and retriever …

luismu opened the pull request:
…implementations

- Implemented the Cloud SQL for PostgreSQL plugin for vector similarity search using pgvector.
- Added installation and configuration instructions in the documentation.
- Created PostgresEngine class with methods for instance creation and vector store table initialization.
- Developed indexer and retriever functionalities for document indexing and retrieval.
- Enhanced Jest configuration for better coverage reporting.
- Added comprehensive tests for engine and indexer functionalities, including error handling.
- Updated package.json to include peer dependencies and improve test command verbosity.
- Introduced utility functions for IAM principal email retrieval.

Description here... Help the reviewer by:
 - linking to an issue that includes more details
 - if it's a new feature include samples of how to use the new feature
 - (optional if issue link is provided) if you fixed a bug include basic bug details

Checklist (if applicable):
- [ ] PR title is following https://www.conventionalcommits.org/en/v1.0.0/
- [x] Tested (manually, unit tested, etc.)
- [ ] Docs updated (updated docs or a docs bug required)


![image](https://github.com/user-attachments/assets/a9c98552-6dfa-4281-a412-e5bb2c77a2b6)

luismu committed:
feat: Add Cloud SQL for PostgreSQL plugin with indexer and retriever implementations

- Implemented the Cloud SQL for PostgreSQL plugin for vector similarity search using pgvector.
- Added installation and configuration instructions in the documentation.
- Created PostgresEngine class with methods for instance creation and vector store table initialization.
- Developed indexer and retriever functionalities for document indexing and retrieval.
- Enhanced Jest configuration for better coverage reporting.
- Added comprehensive tests for engine and indexer functionalities, including error handling.
- Updated package.json to include peer dependencies and improve test command verbosity.
- Introduced utility functions for IAM principal email retrieval.

luismu committed:
feat: Add Cloud SQL for PostgreSQL plugin with indexer and retriever implementations

- Implemented the Cloud SQL for PostgreSQL plugin for vector similarity search using pgvector.
- Added installation and configuration instructions in the documentation.
- Created PostgresEngine class with methods for instance creation and vector store table initialization.
- Developed indexer and retriever functionalities for document indexing and retrieval.
- Enhanced Jest configuration for better coverage reporting.
- Added comprehensive tests for engine and indexer functionalities, including error handling.
- Updated package.json to include peer dependencies and improve test command verbosity.
- Introduced utility functions for IAM principal email retrieval.

averikitsch committed:
Merge branch 'upstream-rebase' into indexer

averikitsch commented:
/gcbrun

averikitsch committed:
Update package.json

averikitsch commented:
/gcbrun

averikitsch committed:
Update jest.config.js

averikitsch commented:
/gcbrun

averikitsch commented:
Please fix the failing tests. Let me know if you can't see the logs. The formatting also needs to be fixed by running `pnpm format`

averikitsch commented on js/plugins/cloud-sql-pg/src/utils/index.ts:
please delete the unused utils.ts file

luismu committed:
refactor: Restructure engine and indexer files for improved organization

- Moved utility functions to a dedicated utils directory for better modularity.
- Updated import paths in engine.ts and index.ts to reflect the new structure.
- Enhanced error messages in index.ts for clarity during document indexing.
- Removed outdated test files and added new integration tests for indexer and retriever functionalities.
- Improved test coverage and organization in the new index.test.ts file.

luismu committed:
Merge branch 'indexer' of https://github.com/luismu/genkit into indexer

luismu committed:
chore: Remove outdated indexer integration test file

- Deleted the indexer.test.ts file as part of cleanup to streamline the test suite.
- This removal helps maintain focus on relevant and up-to-date tests for the indexer functionality.

averikitsch commented:
/gcbrun

averikitsch commented on js/plugins/cloud-sql-pg/jest.config.js:
Remove these changes

luismu replied:
This means line 5, 6, 11, 12 and 13?

averikitsch commented on js/plugins/cloud-sql-pg/package.json:
Remove these changes

luismu replied:
This means remove line 42?

averikitsch commented:
We need to make sure the PostgresEngine is exported correctly

luismu committed:
Refactor code structure for improved readability and maintainability

averikitsch commented:
/gcbrun

averikitsch commented on docs/plugins/cloud-sql-pg.md:
````diff
@@ -0,0 +1,177 @@
+To use this plugin, first create a `PostgresEngine` instance:
+
+```ts
+import { PostgresEngine } from '@genkit-ai/cloud-sql-pg';
````
Is this the right import package?

luismu replied:
fixed.

luismu committed:
formatter prettier

luismu committed:
docs: Update import path for PostgresEngine in Cloud SQL PostgreSQL plugin documentation

luismu committed:
chore: Remove unused utility function for IAM principal email retrieval

luismu committed:
utils/index.ts file

averikitsch commented:
/gcbrun

luismu committed:
format prettier

averikitsch commented:
/gcbrun

averikitsch commented on docs/plugins/cloud-sql-pg.md:
````diff
@@ -0,0 +1,177 @@
+To use this plugin, first create a `PostgresEngine` instance:
+
+```ts
+import { PostgresEngine } from 'genkit/plugins/cloud-sql-pg';
````
```suggestion
import { PostgresEngine } from 'genkitx-cloud-sql-pg';
```

averikitsch commented on docs/plugins/cloud-sql-pg.md:
````diff
@@ -0,0 +1,177 @@
+
+```ts
+import { genkit } from 'genkit';
+import { postgres } from '@genkit-ai/cloud-sql-pg';
````
fix package name

luismu committed:
chore: Update Jest configuration and package dependencies

- Simplified jest.config.js by removing unnecessary options and comments.
- Updated test roots and match patterns for better test discovery.
- Added @types/jest and updated typescript version in package.json.
- Included zod as a new dependency.

luismu committed:
chore: Update package dependencies in cloud-sql-pg

- Updated @types/node to version 20.11.16.

- Added @genkit-ai/core as a peer dependency.

luismu committed:
fix: Correct import paths for PostgresEngine and postgres in cloud-sql-pg documentation

luismu commented:
![image](https://github.com/user-attachments/assets/0b4d3bd4-d846-443b-a9a7-24399b051817)

averikitsch commented:
/gcbrun

averikitsch committed:
Update jest.config.js

averikitsch committed:
Update package.json

averikitsch commented:
/gcbrun

averikitsch committed:
Update engine.test.ts

averikitsch reviewed (commented):
/gcbrun

averikitsch commented:
/gcbrun

averikitsch committed:
Update engine.test.ts

averikitsch commented:
/gcbrun

averikitsch commented:
/gcbrun

averikitsch commented on js/plugins/cloud-sql-pg/src/engine.ts:
```diff
@@ -185,166 +222,170 @@ export class PostgresEngine {
+    const idColumnName =
+      typeof idColumn === 'string' ? idColumn : idColumn.name;
 
     let query = `CREATE TABLE ${schemaName}.${tableName}(
```
Add quotes around schema and table names

averikitsch committed:
update

averikitsch commented:
/gcbrun

luismu committed:
fix: Enclose schema and table names in double quotes in CREATE TABLE query for Postgres compatibility

luismu committed:
Merge branch 'indexer' of https://github.com/luismu/genkit into indexer
